src/components/Ideas/CompIdeaList.js

```diff
@@ -0,0 +1,42 @@
+import React, { useContext, useEffect, useState } from "react"
+import { useHistory } from "react-router-dom"
+import { IdeaContext } from "./IdeaProvider"
+import { CompIdeaDetail } from "./CompIdeaDetail"
+import "./Idea.css"
+
+export const CompIdeaList = () => {
+    const { ideas, getCompleteIdeas, searchTerms } = useContext(IdeaContext)
+    const [ filteredIdeas, setFiltered ] = useState([])
+    const history = useHistory()
+
+    useEffect(() => {getCompleteIdeas()}, [])
+
+    useEffect(() => {
+        if (searchTerms !== "") {
```

Comment:
Since an empty string is falsy, you would only have to do the following

```suggestion
        if (!searchTerms) {
```